### PR TITLE
Flex deploy examples, and miscellaneous documentation updates.

### PR DESCRIPTION
**Pull request by verult**
* Updating documentation to reflect the newly updated Flexvolume example directory. Depends on https://github.com/kubernetes/kubernetes/pull/59149.
* Removing mentions of --enable-controller-attach-detach because setting it to false is not recommended.
* Adding documentation about pod information passed as mounter options.

/sig storage
/assign @chakri-nelluri @saad-ali

### CODE REVIEW

**Comment by verult** [comment]
gentle ping

**Comment by chakri-nelluri** [comment]
@verult we have two more options missing in the doc. 
```
"kubernetes.io/mountsDir"
"kubernetes.io/pvOrVolumeName"
```
Other than that everything looks good to me.

**Comment by verult** [comment]
ACK, although `"kubernetes.io/mountsDir"` isn't used anywhere in code (besides being declared)?

**Comment by chakri-nelluri** [comment]
My bad in copy & paste
```	"kubernetes.io/fsGroup" ```
is the other option missing.

**Comment by verult** [comment]
Addressed comments

File: contributors/devel/flexvolume.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Flexvolume
 
-Flexvolume enables users to write their own drivers and add support for their volumes in Kubernetes. Vendor drivers should be installed in the volume plugin path on every Kubelet node and on master node(s) if `--enable-controller-attach-detach` Kubelet option is enabled.
+Flexvolume enables users to write their own drivers and add support for their volumes in Kubernetes. Vendor drivers should be installed in the volume plugin path on every node, and on master if the driver requires attach capability.
```
Comment (chakri-nelluri) [inline contributors/devel/flexvolume.md]: This statement is not completely true. If the --enable-controller-attach-detach is false, kubelet will do the attach.

Comment (verult) [reply contributors/devel/flexvolume.md]: I think we should strongly discourage users from setting that flag to false; k8s is no longer designed to operate in that mode. Having this flag in the documentation gives users new to k8s the idea that disabling attach detach is a common and viable alternative, but in most cases it's not.

I'm open to wording it differently but I'd avoid including the flag in the documentation

**Comment by verult** [comment]
PTAL

**Comment by chakri-nelluri** [comment]
@verult Did you take a re-stab a the wording? It still sounds the same. I just want to make sure it won't confuse existing users.

**Comment by verult** [comment]
@chakri-nelluri how about that?

**Comment by verult** [comment]
Updated example links to the examples repo.

**Comment by chakri-nelluri** [comment]
@verult Please rebase.

**Comment by verult** [comment]
@chakri-nelluri ACK. @calebamiles PTAL, thanks!

**Pull request merged**